src/System.Text.Json/tests/NewtonsoftTests/EnumConverterTests.cs

```diff
@@ -0,0 +1,539 @@
+                    Value2 = DuplicateNameEnum2.foo_bar_NOT_USED
+                });
+
+                string xml = @"<DuplicateEnumNameTestClass xmlns=""http://schemas.datacontract.org/2004/07/Newtonsoft.Json.Tests.Converters"" xmlns:i=""http://www.w3.org/2001/XMLSchema-instance"">
+    <Value>foo_bar</Value>
+    <Value2>foo_bar</Value2>
+</DuplicateEnumNameTestClass>";
+
+                var o = (DuplicateEnumNameTestClass)s.ReadObject(new MemoryStream(Encoding.UTF8.GetBytes(xml)));
+
+                Assert.Equal(DuplicateNameEnum.foo_bar, o.Value);
+                Assert.Equal(DuplicateNameEnum2.FooBar, o.Value2);
+            });
+            Assert.Equal(e.Message, "Type 'System.Text.Json.Tests.DuplicateNameEnum' contains two members 'foo_bar' 'and 'FooBar' with the same name 'foo_bar'. Multiple members with the same name in one type are not supported. Consider changing one of the member names using EnumMemberAttribute attribute.");
+        }
+
+        [Fact]
+        public static void TestValidIntegerValue()
+        {
+            StoreColor c = JsonSerializer.Parse<StoreColor>("1");
+            Assert.Equal(c, StoreColor.Black);
+        }
+
+        [Fact]
+        public static void DuplicateNameEnumTest()
+        {
+            JsonException e = Assert.Throws<JsonException>(() =>
+                JsonSerializer.Parse<DuplicateNameEnum>("\"foo_bar\""));
+        }
+
+        [Fact]
+        public static void InvalidValueStringNumber()
+        {
+            JsonException ex = Assert.Throws<JsonException>(() =>
+            {
+                StoreColor s = JsonSerializer.Parse<StoreColor>("\"1\"");
+            });
+        }
+
+        [Fact]
+        public static void SerializeEnumWithDifferentCases()
+        {
+            string json = JsonSerializer.ToString(EnumWithDifferentCases.M);
+
+            Assert.Equal("0", json);
+
+            json = JsonSerializer.ToString(EnumWithDifferentCases.m);
+
+            Assert.Equal("1", json);
+        }
+
+        [Fact]
+        public static void DeserializeEnumWithDifferentCases()
+        {
+            EnumWithDifferentCases e = JsonSerializer.Parse<EnumWithDifferentCases>("0");
+            Assert.Equal(EnumWithDifferentCases.M, e);
+
+            e = JsonSerializer.Parse<EnumWithDifferentCases>("1");
+            Assert.Equal(EnumWithDifferentCases.m, e);
+        }
+
+        [Fact]
+        public static void SerializeEnumMemberWithDifferentCases()
+        {
+            string json = JsonSerializer.ToString(EnumMemberWithDifferentCases.Month);
+
+            Assert.Equal("0", json);
+
+            json = JsonSerializer.ToString(EnumMemberWithDifferentCases.Minute);
+
+            Assert.Equal("1", json);
+        }
+
+        [Fact]
+        public static void InvalidValueNegativeStringNumber()
+        {
+            JsonException ex = Assert.Throws<JsonException>(() =>
+            {
+                JsonSerializer.Parse<StoreColor>("\"-1\"");
+            });
```

Comment:
> I believe the "invalid" part of this test is that it's a JSON string token. The negative sign is distracting, and isn't contributing to the invalidness... ((StoreColor)-1) is a valid value since the enum is (implicitly) backed by int.

Yes string bracket makes it invalid

> This leads me to "if we don't already have tests (in the normal suite) de/serializing values from enums whose backing type is something other than int, we should add some"

I didn't really understood your point, but the point of this test is if there is string quote whatever number it had it should fail

Reply:
There are two points here:

1) (Specific to this PR:) I believe this test should be changed to not use a negative number (and should accordingly be renamed).  The method name starts off as "invalid value negative", which suggests (to me) that the negative is what makes it invalid.  "InvalidValueStringNumber" (without the negative) is all that's required to generate the failure case the test is looking for.

2) (General for this library, just a passing comment to anyone currently adding tests to it:) If all of the serializer enum tests use int-backed enums, we need more test diversity.

Reply:
Yes this test doesn't make much sense from the point of the underlying code. if the developer want to check his code working as expected wouldn't write this test by knowing underlying code 

But by my understanding this test is for general case without care of whatever underlying code is. And by my understanding we are going to support string represented numbers in the future in that case this could be testable case

Reply:
And yes `((StoreColor)-1) is a valid value since the enum is (implicitly) backed by int` and this test will fail after string represented number is supported, but for me it is still valid case to test and the test should be updated accordingly that time

Reply:
> (General for this library, just a passing comment to anyone currently adding tests to it:) If all of the serializer enum tests use int-backed enums, we need more test diversity.

cc @maryamariyan, @JeremyKuhne

Reply:
Removed as `"1" and "-1"` test not have different impact and we already have test for "1"